tests/unit/test-cmd/test.build.js

```diff
@@ -329,4 +368,27 @@ describe('build', () => {
 
   });
 
+  describe('defaultPackageCreator', () => {
+
+    it('should reject on Unexpected errors', () => {
+      const fakeEventToPromise = sinon.spy(
+        () => Promise.reject(new Error('Unexpected error')));
+      const params = {
+        manifestData: basicManifest,
+        sourceDir: fixturePath('minimal-web-ext'),
+        fileFilter: {wantFile: () => true},
+        artifactsDir: 'artifacts',
+        eventToPromise: fakeEventToPromise,
```

Comment:
Once you have applied the changes to the "src/cmd/build.js" module described above, we can fix this test accordingly (by moving the fake `eventToPromise` in the second parameter, and by using a real FileFilter instance to prevent flow to complain at line 385):

```diff
diff --git a/tests/unit/test-cmd/test.build.js b/tests/unit/test-cmd/test.build.js
index 14ce59d..ce54181 100644
--- a/tests/unit/test-cmd/test.build.js
+++ b/tests/unit/test-cmd/test.build.js
@@ -373,16 +373,23 @@ describe('build', () => {
     it('should reject on Unexpected errors', () => {
       const fakeEventToPromise = sinon.spy(
         () => Promise.reject(new Error('Unexpected error')));
+      const sourceDir = fixturePath('minimal-web-ext');
+      const artifactsDir = 'artifacts';
+      const fileFilter = new FileFilter({sourceDir, artifactsDir});
       const params = {
         manifestData: basicManifest,
-        sourceDir: fixturePath('minimal-web-ext'),
-        fileFilter: {wantFile: () => true},
-        artifactsDir: 'artifacts',
-        eventToPromise: fakeEventToPromise,
+        sourceDir,
+        fileFilter,
+        artifactsDir,
         overwriteDest: false,
         showReadyMessage: false,
-      };// $FLOW_FIXME: flow complains about the fileFiler type
-      return defaultPackageCreator(params)
+      };
+
+      const options = {
+        eventToPromise: fakeEventToPromise,
+      };
+
+      return defaultPackageCreator(params, options)
         .then(() => assert.fail('The expected error has not been raised'))
         .catch ((error) => {
           assert.match(error.message, /Unexpected error/);
```